instrumentation/othttp/handler.go

```diff
@@ -56,21 +61,26 @@ func NewHandler(handler http.Handler, operation string, opts ...Option) http.Han
 		operation: operation,
 	}
 
+	const domain = "go.opentelemetry.io/otel/instrumentation/othttp"
+
 	defaultOpts := []Option{
-		WithTracer(global.Tracer("go.opentelemetry.io/otel/instrumentation/othttp")),
+		WithTracer(global.Tracer(domain)),
+		WithMeter(global.Meter(domain)),
 		WithPropagators(global.Propagators()),
 		WithSpanOptions(trace.WithSpanKind(trace.SpanKindServer)),
 		WithSpanNameFormatter(defaultHandlerFormatter),
 	}
 
 	c := NewConfig(append(defaultOpts, opts...)...)
 	h.configure(c)
+	h.createMeasures()
 
 	return &h
 }
 
 func (h *Handler) configure(c *Config) {
 	h.tracer = c.Tracer
+	h.meter = metric.Must(c.Meter)
```

Comment:
Shipping instrumentation that will panic is not a great user experience. Can we handle this error with the `global.Handle` or otherwise log a failure?

Reply:
Can this panic at runtime at any point other than measure creation?  My understanding was that the `MeterMust` type would panic rather than returning an error when creating instruments and that's the only time it panics.  If so, it should be the case that this will either always panic or never panic for a given commit and we can identify panics during automated testing.

All of which is a long way of saying I'm not terribly concerned about it, but explicitly handling the errors to avoid a panic would also be reasonable.

Reply:
If the user passes an invalid Meter when they instrument their code this will panic then. While the truth is that kind of error will be the responsibility of the operator, from prior conversations with customers using instrumentation from someone else that panics (even if it is from them misconfiguring things) results in frustration with the design rather than their mistake.

Agreed it's not a huge issue, just an ask.

Reply:
```
would panic rather than returning an error when creating instruments and that's the only time it panics.
```

yep, that was in mind, panic on startup.  didnt see any way to propagate error from handler creation, thought

Reply:
https://github.com/open-telemetry/opentelemetry-go/blob/master/api/global/handler.go#L89-L92

Reply:
examples of use:

- https://github.com/open-telemetry/opentelemetry-go/blob/2966505271c3698127db3d243ff86682e22caff0/sdk/metric/sdk.go#L168
- https://github.com/open-telemetry/opentelemetry-go/blob/2966505271c3698127db3d243ff86682e22caff0/sdk/metric/sdk.go#L178
- https://github.com/open-telemetry/opentelemetry-go/blob/2966505271c3698127db3d243ff86682e22caff0/sdk/metric/controller/push/push.go#L148
- https://github.com/open-telemetry/opentelemetry-go/blob/4bf35c611bdc6498a74f7d1b374ac5cda7508308/exporters/trace/jaeger/jaeger.go#L126

i.e.

```
	handleErr := func(err error) {
		if err != nil {
			global.Handle(err)
		}
	}
	requestCounter, err := h.meter.NewInt64Counter(RequestCount)
	handleErr(err)
...
```

Reply:
thanks, will do